src/callbacks/summary.jl

```diff
@@ -14,42 +22,107 @@ function Base.show(io::IO, cb::DiscreteCallback{Condition,Affect!}) where {Condi
 end
 
 
+function format_key_value_line(key::AbstractString, value::AbstractString, key_width, total_width;
+                               guide='…', filler='…', prefix="│ ", suffix=" │")
+  @assert key_width < total_width
+  line  = prefix
+  squeezed_key = squeeze(key, key_width, filler=filler)
+  line *= squeezed_key
+  line *= ": "
+  short = key_width - length(squeezed_key)
+  if short <= 1
+    line *= " "
+  else
+    line *= guide^(short-1) * " "
+  end
+  value_width = total_width - length(prefix) - length(suffix) - key_width - 2
+  squeezed_value = squeeze(value, value_width, filler=filler)
+  line *= squeezed_value
+  short = value_width - length(squeezed_value)
+  line *= " "^short
+  line *= suffix
+
+  @assert length(line) == total_width "should not happen: algorithm error!"
+
+  return line
+end
+format_key_value_line(key, value, args...; kwargs...) = format_key_value_line(string(key), string(value), args...; kwargs...)
+
+function squeeze(message, max_width; filler='…')
+  @assert max_width >= 3 "squeezing works only for a minimum `max_width` of 3"
+
+  length(message) <= max_width && return message
+
+  keep_front = div(max_width, 2)
+  keep_back = div(max_width, 2) - (isodd(max_width) ? 0 : 1)
+  squeezed = message[begin:keep_front] * filler * message[end-keep_back+1:end]
+
+  @assert length(squeezed) == max_width "should not happen: algorithm error!"
+
+  return squeezed
+end
+
+function boxed_setup(heading, key_width, total_width, setup=[]; guide='…', filler='…')
+  s  = ""
+  s *= "┌" * "─"^(total_width-2) * "┐\n"
+  s *= "│ " * heading * " "^(total_width - length(heading) - 4) * " │\n"
+  # s *= "├" * "─"^(total_width-2) * "┤\n"
+  s *= "│ " * "═"^length(heading) * " "^(total_width - length(heading) - 4) * " │\n"
+  for (key, value) in setup
+    s *= format_key_value_line(key, value, key_width, total_width) * "\n"
+  end
+  s *= "└" * "─"^(total_width-2) * "┘"
+end
+
+
+# Print information about the current simulation setup
+# Note: This is called *after* all initialization is done, but *before* the first time step
 function initialize_summary_callback(cb::DiscreteCallback, u, t, integrator)
 
   mpi_isroot() || return nothing
 
   print_startup_message()
 
   io = stdout
+  key_width = 25
+  total_width = 80 
+  io_context = IOContext(io, :summary => true, :key_width => key_width, :total_width => total_width)
```

Comment:
What about using 
```julia
help?> displaysize
search: displaysize

  displaysize([io::IO]) -> (lines, columns)

  Return the nominal size of the screen that may be used for rendering output to this IO object. If no input is
  provided, the environment variables LINES and COLUMNS are read. If those are not set, a default size of (24,
  80) is returned.
[...]
```
to get the total width and `:displaysize` instead of `:total_width` as argument in the `IOContext`, since that's one of the common arguments?
```julia
help?> IOContext
[....]
  The following properties are in common use:

    •    :compact: Boolean specifying that values should be printed more compactly, e.g. that numbers should
        be printed with fewer digits. This is set when printing array elements. :compact output should not
        contain line breaks.

    •    :limit: Boolean specifying that containers should be truncated, e.g. showing … in place of most
        elements.

    •    :displaysize: A Tuple{Int,Int} giving the size in rows and columns to use for text output. This can
        be used to override the display size for called functions, but to get the size of the screen use
        the displaysize function.
[...]
```
Similarly, I would propose to use `:compact` instead of `:summary` (with flipped meaning, i.e. `get(io, :compact) == !get(io, :summary)`).

Reply:
Re/`displaysize`: I specifically used a fixed value that is independent of the `displaysize` since I wanted the output to be reproducible on different machines and between different users (such that it is easy to, e.g., create a diff between two outputs). We could discuss , however, if we want to increase the total width to 100 (which is the same as for our code) to have more screen space.

Reply:
Re/`:compact`: To me, using `:compact` feels very counter-intuitive, since the summary is by its nature rather verbose, not compact (IMHO). Also, the would be counter the specific suggestion in the docs that
> `:compact` output should not contain line breaks.

Maybe `summary` is also a bad name - should we make it more specific to this use case, such as `simulation_summary`, or more functional, such as `pretty_summary`? (Or without `summary` at all? I am not married to the word here)

Reply:
I'm not quite 100% sure why you want the output to be reproducible. However, increasing the default value to 100 sounds good to me.

Reply:
The one-line output we had before (using the 2-arg `show` method) could be our default value for `get(io, :compact) == true` while your new boxed version would be the version for `get(io, :compact) == false`?

Reply:
Ah, ok. You mean I should just remove "your" output that we used before and make it the default?

> I'm not quite 100% sure why you want the output to be reproducible.

That's a nice feature to have when you run on supercomputers. But more importantly, the boxes look very ugly if they get too wide...

> However, increasing the default value to 100 sounds good to me.

I'll do that. In that case we should at some point also increase the analysis output to 100 characters.

Reply:
> Ah, ok. You mean I should just remove "your" output that we used before and make it the default?

Yes :+1:

Reply:
> I'll do that. In that case we should at some point also increase the analysis output to 100 characters.

:+1:

Reply:
Output width increased to 100: fixed in d92bb51609b8cce783f329eebd4f85042528b309
Use `:compact` instead of `:summary`: fixed in a1d97e6c650531daabab840e9211bde2bd06c76b